File: grind75/Dec21/383.py
from collections import Counter
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        c_r = Counter(ransomNote)
        c_m = Counter(magazine)

        for key in c_r.keys():
            if key not in c_m or c_m[key] < c_r[key]:
                return False
        
        return True

File: grind75/Dec21/383.go
func canConstruct(ransomNote string, magazine string) bool {
    m := [26]int{}
    for _, c := range magazine {
        m[int(c-'a')] += 1
    }
    for _, c := range ransomNote {
        idx := int(c-'a')
        if m[idx] <= 0 { return false}
        m[idx] -= 1
    }
    return true
}
